pkg/fetch/vendir.go

```diff
@@ -421,3 +421,20 @@ func ExtractImageRegistry(name string) string {
 	}
 	return registry
 }
+
+// extractGitHostname extracts the hostname from the git URL.
+func extractGitHostname(input string) string {
+	u, err := url.Parse(input)
+	if err != nil {
+		return "github.com"
+	}
+	return u.Hostname()
+}
+
+// ExtractHost return registry for Docker Image and Host for git url
+func ExtractHost(input string, isGitURL bool) string {
```

Comment:
`isGirURL` doesn't sound very extensible, let's have a _type_ defined for the source type (like imgpkg or git) which we can pass down, and have switch case to extract the host, this way we can extend the functionality for other sources in future if needed.